rerun_cpp/src/rerun/components/half_extents2d_ext.cpp

```diff
@@ -0,0 +1,31 @@
+#include "half_extents2d.hpp"
+
+// Uncomment for better auto-complete while editing the extension.
+// #define EDIT_EXTENSION
+
+namespace rerun {
+    namespace components {
+
+#ifdef EDIT_EXTENSION
+        struct HalfExtents2DExt {
+            float xy[2];
+#define HalfExtents2D HalfExtents2DExt
+
+            // [CODEGEN COPY TO HEADER START]
+
+            /// Construct HalfExtents2D from x/y values.
+            HalfExtents2D(float x, float y) : xy{x, y} {}
+
+            float x() const {
+                return xy.x();
+            }
+
+            float y() const {
```

Comment:
doesn't these require `inline` to work properly in C++? I forget.

Reply:
everything that is defined within a class definition is automatically inline. Some people do like adding the keyword anyways, but it has no effect.